libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/NavigationRouteAlternativesRequestCallback.kt

```diff
@@ -0,0 +1,53 @@
+package com.mapbox.navigation.core.routealternatives
+
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+
+/**
+ * Interface definition for a callback that is notified whenever alternatives routes are refreshed on demand.
+ *
+ * See [MapboxNavigation.requestAlternativeRoutes].
+ */
+interface NavigationRouteAlternativesRequestCallback {
+
+    /**
+     * Invoked when on-demand alternative routes request finishes.
+     *
+     * The [alternatives] list always represent all available, up-to-date, alternatives for the current route.
+     *
+     * The alternatives are not automatically added to [MapboxNavigation],
+     * you need to add them manually to trigger [RoutesObserver], for example:
+     * todo update docs (no fun int)
```

Comment:
Should we cut tickets for the `TODO`s so we don't forget?

Reply:
Fixed.